pkg/apis/imageregistry/v1/types.go

```diff
@@ -99,52 +99,57 @@ type ConfigList struct {
 type Config struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
-	Spec              ImageRegistrySpec   `json:"spec"`
-	Status            ImageRegistryStatus `json:"status"`
+
+	Spec ImageRegistrySpec `json:"spec"`
+	// +optional
+	Status ImageRegistryStatus `json:"status"`
 }
 
 type ImageRegistrySpec struct {
 	// ManagementState indicates whether the registry instance represented
 	// by this config instance is under operator management or not.  Valid
-	// values are Managed, Unmanaged, and Removed
+	// values are Managed, Unmanaged, and Removed.
 	ManagementState operatorsv1api.ManagementState `json:"managementState"`
-
-	// HTTPSecret is the value needed by the registry to secure uploads, generated by default
+	// HTTPSecret is the value needed by the registry to secure uploads, generated by default.
+	// +optional
 	HTTPSecret string `json:"httpSecret"`
-
-	// Proxy defines the proxy to be used when calling master api, upstream registries, etc
+	// Proxy defines the proxy to be used when calling master api, upstream
+	// registries, etc.
+	// +optional
 	Proxy ImageRegistryConfigProxy `json:"proxy"`
```

Comment:
Need a pointer here so we can distinguish between "no proxy was set" vs. "don't use a proxy for the registry".

```suggestion
	Proxy *ImageRegistryConfigProxy `json:"proxy,omitempty"`
```

Reply:
Do you want this PR to change the logic for proxies? Or should this PR depend on your PR about a generic Proxy?

Reply:
@adambkaplan ?